ReplayKitExample/ReplayKitExample/ReplayKitVideoSource.swift

```diff
@@ -210,6 +228,40 @@ class ReplayKitVideoSource: NSObject, TVIVideoCapturer {
                      orientation: videoOrientation)
     }
 
+    func preallocateScalingBuffers(sourceY: UnsafePointer<vImage_Buffer>,
+                                   sourceUV: UnsafePointer<vImage_Buffer>,
+                                   destinationY: UnsafePointer<vImage_Buffer>,
+                                   destinationUV: UnsafePointer<vImage_Buffer>) {
+        // Size the buffers required for vImage scaling. As source requirements change, we might need to reallocate them.
+        let yBufferSize = vImageScale_Planar8(sourceY, destinationY, nil, vImage_Flags(kvImageGetTempBufferSize))
+        assert(yBufferSize > 0)
+
+        if (downscaleYPlaneBuffer == nil) {
+            downscaleYPlaneBuffer = malloc(yBufferSize)
```

Comment:
do you think we should initialize the memory with `0` ?

Reply:
I will double check if its needed by vImage. If not, I would prefer to skip it.